database-commons/src/main/java/io/cdap/plugin/db/sink/ETLDBOutputFormat.java

```diff
@@ -208,4 +230,45 @@ public String constructQuery(String table, String[] fieldNames) {
     }
     return query;
   }
+  public String constructQueryUpdateAndUpsert(String tableName, String[] fieldNames, String operationName,
+                                              String[] listKeys) {
+    String query = null;
+    if (operationName.equalsIgnoreCase(UPDATE)) {
+      query = constructUpdateQuery(tableName, fieldNames, listKeys);
+    } else if (operationName.equalsIgnoreCase(UPSERT)) {
+      query = constructUpsertQuery(tableName, fieldNames, listKeys);
+    }
+    return query;
+  }
```

Comment:
Add a space. if these methods are not getting invoked outside this class, make it private or protected. Also write comments on top of it.

Reply:
this method is being used outside of this class, therefore public.

Reply:
done